### ISSUE TITLE
Add keyframes-name-pattern

### ISSUE THREAD

**Issue by tlenex** [stylelint/stylelint#3262]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I'd like to make an Feature request, where `@keyframes` definition's name needs to match given pattern, something like `selector-class-pattern` rule, but for keyframes.

I'd like to use an namespace validation rule for all my keyframe definitions.

> Which rule, if any, is this issue related to?

Issue relates to similar rule `selector-class-pattern`.

> What CSS is needed to reproduce this issue?

```css
@keyframes NAMESPACE-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

n/a

> Which version of stylelint are you using?

`9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config myconfig *.css` should be ok.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

I expect warnings to be flagged if keyframe definition name doesn't match the pattern.

> What actually happened (e.g. what warnings or errors you are getting)?

Nothing related to the feature request, because of lacking rules.

**Comment by evilebottnawi** [comment]
Sound good to me :+1:

**Comment by jeddy3** [comment]
@tlenex Thanks for the request and for using the template.

As @evilebottnawi said, this sounds like a good rule. It's fulfils our [criteria for inclusion](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#criteria-for-inclusion).

Please consider [contributing](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#creating-a-new-rule) it if you'll find it useful.

- Name: `keyframes-name-pattern` \*
- Description: Specify a pattern for keyframes names
- Primary option: `string/regex`
- Secondary options: none
- Message: "Expected keyframes name to match specified pattern"
- Section: "Limit language features -> Keyframes"

\* That name is most aligned with existing rules (e.g. `media-feature-name-case` etc) and in line with the [spec](https://www.w3.org/TR/css-animations-1/#keyframes). As an aside, the `custom-media-pattern` rule [should have been called](https://drafts.csswg.org/mediaqueries-5/#custom-mq) `custom-media-name-pattern`, but that ship has long sailed.

**Comment by MatthiasKunnen** [comment]
@tlenex, are you working on this rule? I could take a look.

**Comment by tlenex** [comment]
@MatthiasKunnen unfortunately I don't have time to pull this.
